include/envoy/event/file_event.h

```diff
@@ -18,15 +18,24 @@ struct FileReadyType {
   static const uint32_t Closed = 0x4;
 };
 
-enum class FileTriggerType { Level, Edge };
+#define FORCE_LEVEL_EVENTS
+enum class FileTriggerType { Level, Edge, EmulatedEdge };
 
-static constexpr FileTriggerType PlatformDefaultTriggerType
-#ifdef WIN32
-    // Libevent only supports Level trigger on Windows.
-    {FileTriggerType::Level};
+static constexpr bool optimizeLevelEvents = true;
```

Comment:
Should optimizeLevelEvents be protected by a runtime feature or compile time define?

I guess the question is wherever or not we're comfortable making EmulatedEdge the WIN32 default now or in a followup PR.  I don't have a strong opinion on this, the improvements introduced by EmulatedEdge address a major issue.

Use of a runtime feature seems like a good option since it allows control without recompile.  Of course we'ld need some separation between PlatformDefaultTriggerType which needs to be constexpr and an accessor function that depends on runtime features used only at the time when socket iohandles are created.

Reply:
See https://github.com/envoyproxy/envoy/pull/13787#issuecomment-729884047